rocAL/rocAL_pybind/amd/rocal/plugin/generic.py

```diff
@@ -22,66 +22,80 @@
 import numpy as np
 import rocal_pybind as b
 import amd.rocal.types as types
-import ctypes
+
 
 class ROCALGenericIterator(object):
-    def __init__(self, pipeline, tensor_layout = types.NCHW, reverse_channels = False, multiplier = [1.0,1.0,1.0], offset = [0.0, 0.0, 0.0], tensor_dtype=types.FLOAT, display=False, device="cpu", device_id =0):
+    def __init__(self, pipeline, tensor_layout=types.NCHW, reverse_channels=False, multiplier=[1.0, 1.0, 1.0], offset=[0.0, 0.0, 0.0], tensor_dtype=types.FLOAT, display=False, device="cpu", device_id=0):
         self.loader = pipeline
         self.tensor_format = tensor_layout
         self.multiplier = multiplier
         self.offset = offset
-        self.device= device
+        self.device = device
         self.device_id = device_id
         self.reverse_channels = reverse_channels
         self.tensor_dtype = tensor_dtype
         self.display = display
         self.batch_size = pipeline._batch_size
         if self.loader._name is None:
             self.loader._name = self.loader._reader
-        self.labels_size = ((self.bs*self.loader._numOfClasses) if (self.loader._oneHotEncoding == True) else self.bs)
-        self.out = self.dimensions = self.dtype = None
-        self.len = b.getRemainingImages(self.loader._handle)//self.bs # iteration length
+        self.labels_size = ((self.batch_size * self.loader._num_classes) if self.loader._one_hot_encoding else self.batch_size)
+        self.output_list = self.dimensions = self.dtype = None
+        self.labels_tensor = None
+        self.len = b.getRemainingImages(self.loader._handle) // self.batch_size  # iteration length
 
     def next(self):
         return self.__next__()
 
     def __next__(self):
-        
-        if(b.isEmpty(self.loader._handle)):
+
+        if self.loader.rocalRun() != 0:
             raise StopIteration
         else:
```

Comment:
I think we can remove this else condition here

If this condition is satisfied, we anyways are going to StopIteration right ?